src/diffusers/models/unets/unet_2d.py

```diff
@@ -194,19 +195,22 @@ def __init__(
             self.down_blocks.append(down_block)
 
         # mid
-        self.mid_block = UNetMidBlock2D(
-            in_channels=block_out_channels[-1],
-            temb_channels=time_embed_dim,
-            dropout=dropout,
-            resnet_eps=norm_eps,
-            resnet_act_fn=act_fn,
-            output_scale_factor=mid_block_scale_factor,
-            resnet_time_scale_shift=resnet_time_scale_shift,
-            attention_head_dim=attention_head_dim if attention_head_dim is not None else block_out_channels[-1],
-            resnet_groups=norm_num_groups,
-            attn_groups=attn_norm_num_groups,
-            add_attention=add_attention,
-        )
+        if mid_block_type is None:
+            self.mid_block = None
+        else:
```

Comment:
Hmm, if it's unsupported then should we not be disallowing its usage? I understand that's a breaking change and in that case, a deprecation would be preferred. WDYT?

Reply:
If we don't want to support `mid_block_type=None` then we can just remove the docstring, the argument itself currently doesn't exist.

https://github.com/huggingface/diffusers/blob/b5726358cf125f2fa1a596dce321e91a225a57e4/src/diffusers/models/unets/unet_2d_condition.py#L1244

We support `mid_block_type` `None` in `UNet2DConditionModel` so it seemed appropriate to support in `UNet2DModel`.

We could use `get_mid_block` here but simpler like this as we'd need to guard against the other 2 `CrossAttn` mid blocks.

Reply:
> We support mid_block_type None in UNet2DConditionModel so it seemed appropriate to support in UNet2DModel.

I think since it's unsupported anyway in `UNet2DModel`, I am not seeing the reason to pass a `None` but if it's simpler to do so maintenance-wise, I am okay with that,

Reply:
See the linked issue https://github.com/huggingface/diffusers/issues/10460
```python
>>> import diffusers
>>> diffusers.UNet2DModel(mid_block_type=None)
```

Reply:
Okay works for me then but I will note that supporting just `None` values does feel weird to me.